app/src/main/java/com/vultisig/wallet/ui/screens/NamingVaultScreen.kt

```diff
@@ -0,0 +1,161 @@
+package com.vultisig.wallet.ui.screens
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text2.BasicTextField2
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.KeyboardArrowLeft
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import com.vultisig.wallet.R
+import com.vultisig.wallet.ui.components.MultiColorButton
+import com.vultisig.wallet.ui.models.NamingVaultViewModel
+import com.vultisig.wallet.ui.theme.Theme
+import com.vultisig.wallet.ui.theme.dimens
+
+@OptIn(ExperimentalFoundationApi::class, ExperimentalMaterial3Api::class)
+@Composable
+internal fun NamingVaultScreen(
+    navController: NavHostController,
+) {
+    val viewModel = hiltViewModel<NamingVaultViewModel>()
+    val uiModel by viewModel.uiModel.collectAsState()
+    val snackBarHostState = remember { SnackbarHostState() }
+    LaunchedEffect(key1 = Unit) {
+        viewModel.snackBarChannel.collect { message ->
+            message?.let {
+                snackBarHostState.showSnackbar(message)
+            }
+        }
+    }
```

Comment:
I don't see any reason to duplicate this code. There's VaultRenameScreen which is basically the same

Reply:
Done

Reply:
I don't see any improvements about it, NamingVaultScreen still seems to be a duplicate of VaultRenameScreen

Reply:
@YGolybev 
only ui of two screens are same. but the arguments of destinations and the functionality of the screens differ from each other. if i merge them into one screen the functionality of that will lost its single responsibility and make its viewmodel more complex and unclear.

Reply:
@YGolybev 

only ui of two screens are same. but the arguments of destinations and the functionality of the screens differ from each other. if i merge them into one screen the functionality of that will lost its single responsibility and make its viewmodel more complex and unclear.

Reply:
You can reuse UI with different ViewModels. There's even 

```
private fun VaultRenameScreen(
    navHostController: NavHostController,
    uiModel: VaultEditUiModel,
    snackBarHostState: SnackbarHostState,
    onSave: () -> Unit = {},
    onChangeName: (name: String) -> Unit = {},
)
```

for that case already.